### PR TITLE
[cleanup] silence compiler warnings

### PR DESCRIPTION
**Pull request by Rechi**
## Description
Silence compiler warnings.

## Motivation and Context
I don't like those warnings.

## How Has This Been Tested?
compiled for osx and windows

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/RepositoryUpdater.cpp [hunk 1]
```diff
@@ -43,8 +43,7 @@ namespace ADDON
 
 CRepositoryUpdater::CRepositoryUpdater(CAddonMgr& addonMgr) :
   m_timer(this),
-  m_doneEvent(true),
-  m_addonMgr(addonMgr)
```
Comment (garbear) [inline xbmc/addons/RepositoryUpdater.cpp]: This variable should be used below instead of `CAddonMgr::GetInstance()` I'm guessing

Comment (Rechi) [inline xbmc/addons/RepositoryUpdater.cpp]: @tamland you added it at #12427, but it doesn't get used anywhere

Comment (tamland) [inline xbmc/addons/RepositoryUpdater.cpp]: Yes, as @garbear said. Just sloppiness on my part..

Comment (Rechi) [inline xbmc/addons/RepositoryUpdater.cpp]: @tamland I'll drop the change
please create a seperate PR

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp [hunk 2]
```diff
@@ -29,8 +29,6 @@
 
 using namespace kodi::addon;
 
-#define ALIGN(value, alignment) (((value)+(alignment-1))&~(alignment-1))
-
```
Comment (Rechi) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp]: @FernetMenta is it okay for you to change this here or do you want me to PR it to vpupdates?

Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp]: ok, change it here

**Comment by Rechi** [comment]
just a random test suite failure on linux

**Pull request merged**